### PR TITLE
console: mark inconsistent schemas in the UI (close #5093)

### LINKED ISSUE #5093: console: mark inconsistent remote schemas in the UI
Currently, we [filter out inconsistent remote schemas](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/RemoteSchema/Actions.js#L58) and don't display them in the UI. 

Instead, we can display them on the list with a warning symbol.

### PR DESCRIPTION
**Pull request by kolharsam**
closes #5093 

### Description
Inconsistent Remote Schemas are not rendered on the `Remote Schema` page on updating metadata. This PR fixes that and allows users to modify the inconsistent schema from the `Remote Schema` page

### Changelog (required)

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [x] Console

### CODE REVIEW

File: console/src/components/Services/RemoteSchema/Actions.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { loadMigrationStatus } from '../../Main/Actions';
 import { handleMigrationErrors } from '../../../utils/migration';
 
 import { showSuccessNotification } from '../Common/Notification';
-import { filterInconsistentMetadataObjects } from '../Settings/utils';
+// import { filterInconsistentMetadataObjects } from '../Settings/utils';
```
Comment (kolharsam) [inline console/src/components/Services/RemoteSchema/Actions.js:15]: Remove comment

File: console/src/components/Services/Settings/MetadataOptions/ReloadRemoteSchema.js [hunk 2]
```diff
@@ -15,27 +17,38 @@ class ReloadRemoteSchema extends Component {
...
-            dispatch(
-              showErrorNotification(
-                'Error reloading remote schema',
-                null,
-                error
-              )
-            );
-            this.setState({ isReloading: false });
+          () => {
+            showErrorNotification('Error reloading remote schema', null);
```
Comment (kolharsam) [inline console/src/components/Services/Settings/MetadataOptions/ReloadRemoteSchema.js:51]: Need to discuss this with @beerose

File: console/src/components/Services/Settings/MetadataOptions/ReloadRemoteSchema.js [hunk 3]
```diff
@@ -15,27 +17,38 @@ class ReloadRemoteSchema extends Component {
     this.state.isReloading = false;
   }
   render() {
-    const { dispatch, remoteSchemaName } = this.props;
+    const { dispatch } = this.props;
+    console.group(this.props);
```
Comment (kolharsam) [inline console/src/components/Services/Settings/MetadataOptions/ReloadRemoteSchema.js:21]: 😱

File: console/src/components/Services/Settings/MetadataOptions/ReloadRemoteSchema.js (hunk 2 above)
Comment (kolharsam) [reply console/src/components/Services/Settings/MetadataOptions/ReloadRemoteSchema.js]: fixed this!

File: console/src/components/Services/RemoteSchema/Edit/View.js [hunk 4]
```diff
@@ -17,9 +17,9 @@ import ReloadRemoteSchema from '../../Settings/MetadataOptions/ReloadRemoteSchem
 
 import { appPrefix } from '../constants';
 
-import { NotFoundError } from '../../../Error/PageNotFound';
-
+// import { NotFoundError } from '../../../Error/PageNotFound';
```
Comment (kolharsam) [inline console/src/components/Services/RemoteSchema/Edit/View.js:20]: Remove this.

File: console/src/components/Services/RemoteSchema/Edit/View.js [hunk 5]
```diff
@@ -17,9 +17,9 @@ import ReloadRemoteSchema from '../../Settings/MetadataOptions/ReloadRemoteSchem
```
Comment (kolharsam) [inline console/src/components/Services/RemoteSchema/Edit/View.js:17]: I noticed that a part of this file uses `UNSAFE_componentWillReceiveProps` shall I move that code to `getDerivedStateFromProps` or any other permissible lifecycle method?////

Comment (kolharsam) [reply console/src/components/Services/RemoteSchema/Edit/View.js]: I have changed them in a couple of places relevant to this PR. There are 4 other such instances in the codebase, I can change those too and update them in the same PR or I could make another PR for those too

File: console/src/components/Common/Tooltip/Tooltip.tsx [hunk 6]
```diff
@@ -11,18 +11,20 @@ export interface TooltipProps extends React.ComponentProps<'i'> {
...
   placement = 'right',
   tooltipStyle = '',
   children,
+  icon = 'fa-info-circle',
 }) => (
   <OverlayTrigger placement={placement} overlay={tooltipGen(message)}>
     {children || (
       <i
-        className={`fa fa-info-circle ${styles.tooltipIcon} ${tooltipStyle}`}
+        className={`fa ${icon} ${styles.tooltipIcon} ${tooltipStyle}`}
```
Comment (beerose) [inline console/src/components/Common/Tooltip/Tooltip.tsx:27]: Why this change?

File: console/src/components/Services/RemoteSchema/Edit/View.js [hunk 7]
```diff
@@ -1,28 +1,60 @@
...
+              index !== data.length - 1 ? <hr /> : null,
+            ])}
+      </td>
+    </tr>
+  ) : null;
+
+const RSReloadSchema = ({ readOnlyMode, remoteSchemaName, ...props }) =>
+  !readOnlyMode && remoteSchemaName ? (
+    <div className={`${styles.commonBtn} ${styles.detailsRefreshButton}`}>
+      <span>
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/Edit/View.js:45]: Are those `spans` needed? Here and below

File: console/src/components/Services/RemoteSchema/Edit/View.js [hunk 8]
```diff
@@ -1,28 +1,60 @@
...
+        <ReloadRemoteSchema {...props} remoteSchemaName={remoteSchemaName} />
+      </span>
+      <span>
+        <ToolTip
+          icon="fa-question-circle"
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/Edit/View.js:50]: We had a hardcoded icon in `Tooltip` for consistency. Can you revert this change in `Tooltip`?

File: console/src/components/Services/RemoteSchema/Edit/View.js [hunk 9]
```diff
@@ -1,28 +1,60 @@
...
+    <tr>
+      <td>Headers</td>
+      <td>
+        {data &&
+          data
+            .filter(header => !!header.name)
+            .map((header, index) => [
+              <tr key={header}>
+                <td>
+                  {header.name}:{' '}
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/Edit/View.js:30]: Could you use string template here?

File: console/src/components/Common/Tooltip/Tooltip.tsx (hunk 6 above)
Comment (kolharsam) [reply console/src/components/Common/Tooltip/Tooltip.tsx:27]: I made this change because we need the tooltip to be present over a different icon (the exclamation mark) than the default icon

File: console/src/components/Services/RemoteSchema/Edit/View.js (hunk 7 above)
Comment (kolharsam) [reply console/src/components/Services/RemoteSchema/Edit/View.js:45]: I shall remove them

File: console/src/components/Common/Tooltip/Tooltip.tsx (hunk 6 above)
Comment (beerose) [reply console/src/components/Common/Tooltip/Tooltip.tsx:27]: The idea behind this was to be consistent and always use the same icon for tooltip.

File: console/src/components/Services/RemoteSchema/Add/addRemoteSchemaReducer.js [hunk 10]
```diff
@@ -345,9 +344,11 @@ const modifyRemoteSchema = () => {
       },
     };
 
-    resolveObj.definition.headers = [
-      ...getReqHeader(getState().remoteSchemas.headerData.headers),
-    ];
+    const requiredHeaders = getReqHeader(
+      getState().remoteSchemas.headerData.headers
+    );
+    resolveObj.definition.headers = requiredHeaders;
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/Add/addRemoteSchemaReducer.js:350]: Can we inline this?

File: console/src/components/Services/RemoteSchema/RemoteSchemaSubSidebar.js [hunk 11]
```diff
@@ -46,34 +52,55 @@ const RemoteSchemaSubSidebar = ({
...
+            d.name === viewRemoteSchema &&
+            location.pathname.includes(viewRemoteSchema)
+          ) {
+            activeTableClass = styles.activeLink;
+          }
+
+          if (filteredList.length) {
+            isSchemaInconsistent = true;
+            inconsistentCurrentSchema = filteredList[0];
+          }
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/RemoteSchemaSubSidebar.js:74]: You can use `find` instead of filter. 

Also, let's use only one variable for this:

```suggestion
          let activeTableClass = '';
          if (
            d.name === viewRemoteSchema &&
            location.pathname.includes(viewRemoteSchema)
          ) {
            activeTableClass = styles.activeLink;
          }

          const inconsistentSchema = inconsistentRemoteSchemas.find(elem => elem.definition.name === d.name)
```

And then use `inconsistentSchema` to do:

```js
   {inconsistentSchema ? (
                  <ToolTip message={inconsistentSchema.reason}>
                     // ....
                  </ToolTip>
                ) : null}
```

File: console/src/components/Services/RemoteSchema/RemoteSchemaSubSidebar.js [hunk 12]
```diff
@@ -46,34 +52,55 @@ const RemoteSchemaSubSidebar = ({
...
+                to={`${appPrefix}/manage/${d.name}/details`}
+                data-test={d.name}
+              >
+                <i
+                  className={`${styles.tableIcon} fa fa-code-fork`}
+                  aria-hidden="true"
+                />
+                {d.name}
+                {isSchemaInconsistent ? (
+                  <ToolTip message={inconsistentCurrentSchema.reason}>
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/RemoteSchemaSubSidebar.js:92]: Use `WarningSymbol` instead.

File: console/src/components/Common/Tooltip/Tooltip.tsx (hunk 6 above)
Comment (kolharsam) [reply console/src/components/Common/Tooltip/Tooltip.tsx:27]: I've reverted this change

File: console/src/components/Services/RemoteSchema/Edit/View.js [hunk 13]
```diff
@@ -1,28 +1,55 @@
...
+                </td>
+              </tr>,
+              index !== data.length - 1 ? <hr /> : null,
+            ])}
+      </td>
+    </tr>
+  ) : null;
+
+const RSReloadSchema = ({ readOnlyMode, remoteSchemaName, ...props }) =>
+  !readOnlyMode && remoteSchemaName ? (
```
Comment (beerose) [inline console/src/components/Services/RemoteSchema/Edit/View.js:43]: Previous logic was to check `remoteSchemaName` length as well.

**Comment by beerose** [comment]
Please apply the same padding as in Data section. Also, I guess something went wrong with the tooltip.
Screenshot 2020-07-10 at 15 43 59

File: console/src/components/Services/RemoteSchema/Edit/Edit.js [hunk 14]
```diff
@@ -242,6 +241,10 @@ class Edit extends React.Component {
           >
             <Common {...this.props} />
             {generateMigrateBtns()}
+            <small>
+              Note: Please make sure to rectify the errors present in the schema
+              before trying to modify it's details.
+            </small>
```
Comment (kolharsam) [inline console/src/components/Services/RemoteSchema/Edit/Edit.js:247]: @beerose added this note to sort of mitigate the problem that arises when we try to modify inconsistent schemas. I'll remove this if it is not required.

Comment (beerose) [reply console/src/components/Services/RemoteSchema/Edit/Edit.js:247]: I think it should only be present if remote schema is inconsistent. It may be confusing IMO.

**Review by rikinsk — APPROVED** [review]
UX approved

File: console/src/components/Services/RemoteSchema/Edit/Edit.js (hunk 14 above)
Comment (rikinsk) [reply console/src/components/Services/RemoteSchema/Edit/Edit.js:247]: Updated this to be display msg on details page and disabling modify/delete btns

**Review by beerose — APPROVED** [review]
